source/common/router/rds_impl.h

```diff
@@ -164,6 +169,15 @@ class RdsRouteConfigSubscription : Envoy::Config::SubscriptionCallbacks,
 
 using RdsRouteConfigSubscriptionSharedPtr = std::shared_ptr<RdsRouteConfigSubscription>;
 
+struct UpdateOnDemandCallback {
+  const std::set<std::string> aliases_;
```

Comment:
The const give me the hint that the set should be a shared_ptr among all threads